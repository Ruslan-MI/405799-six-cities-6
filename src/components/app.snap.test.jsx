import React from "react";
import {
  render,
} from "@testing-library/react";
import * as redux from "react-redux";
import App from "./app";
import {
  AppRoute,
  StoreNameSpace,
} from "../const";
import Test, {
  mockDataInitialState,
  mockUserInitialState,
} from "../mocks-for-test";

describe(`Test routing`, () => {
  const dataForStore = {
    [StoreNameSpace.DATA]: mockDataInitialState,
    [StoreNameSpace.USER]: mockUserInitialState,
  };

  jest.spyOn(redux, `useSelector`);
  jest.spyOn(redux, `useDispatch`);

  it(`Render main page when user navigate to "/" url`, () => {
    const {
      container,
    } = render(<Test dataForStore={dataForStore}>
      <App />
    </Test>);

    expect(container).toMatchSnapshot();
  });

  it(`Render login page when user navigate to "/login" url`, () => {
    const {
      container,
    } = render(<Test dataForStore={dataForStore} pushURL={AppRoute.LOGIN}>
      <App />
    </Test>);

    expect(container).toMatchSnapshot();
  });

  it(`Render "Not found" page when user navigate to non-existent route`, () => {
    const {
      container,
    } = render(<Test dataForStore={dataForStore} pushURL={`/non-existent-route`}>
      <App />
    </Test>);

    expect(container).toMatchSnapshot();
  });
});
