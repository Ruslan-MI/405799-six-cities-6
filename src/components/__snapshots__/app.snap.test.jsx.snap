// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test routing Render "Not found" page when user navigate to non-existent route 1`] = `
<div>
  <div
    class="page page--gray page--not-found"
  >
    <header
      class="header"
    >
      <div
        class="container"
      >
        <div
          class="header__wrapper"
        >
          <div
            class="header__left"
          >
            <a
              class="header__logo-link"
              href="/"
            >
              <img
                alt="6 cities logo"
                class="header__logo"
                height="41"
                src="img/logo.svg"
                width="81"
              />
            </a>
          </div>
          <nav
            class="header__nav"
          >
            <ul
              class="header__nav-list"
              style="flex-direction: column; align-items: flex-end;"
            >
              <li
                class="header__nav-item user"
              >
                <a
                  class="header__nav-link header__nav-link--profile"
                  href="/login"
                >
                  <div
                    class="header__avatar-wrapper user__avatar-wrapper"
                  />
                  <span
                    class="header__login"
                  >
                    Sign in
                  </span>
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </header>
    <main
      class="page__main page__main--not-found"
    >
      <h1
        style="text-align: center;"
      >
        404 Not Found
      </h1>
    </main>
  </div>
</div>
`;

exports[`Test routing Render login page when user navigate to "/login" url 1`] = `
<div>
  <div
    class="page page--gray page--login"
  >
    <header
      class="header"
    >
      <div
        class="container"
      >
        <div
          class="header__wrapper"
        >
          <div
            class="header__left"
          >
            <a
              class="header__logo-link"
              href="/"
            >
              <img
                alt="6 cities logo"
                class="header__logo"
                height="41"
                src="img/logo.svg"
                width="81"
              />
            </a>
          </div>
          <nav
            class="header__nav"
          >
            <ul
              class="header__nav-list"
              style="flex-direction: column; align-items: flex-end;"
            >
              <li
                class="header__nav-item user"
              >
                <a
                  class="header__nav-link header__nav-link--profile"
                  href="/login"
                >
                  <div
                    class="header__avatar-wrapper user__avatar-wrapper"
                  />
                  <span
                    class="header__login"
                  >
                    Sign in
                  </span>
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </header>
    <main
      class="page__main page__main--login"
    >
      <div
        class="page__login-container container"
      >
        <section
          class="login"
        >
          <h1
            class="login__title"
          >
            Sign in
          </h1>
          <form
            action="#"
            class="login__form form"
            method="post"
          >
            <div
              class="login__input-wrapper form__input-wrapper"
            >
              <label
                class="visually-hidden"
              >
                E-mail
              </label>
              <input
                class="login__input form__input"
                name="email"
                placeholder="Email"
                required=""
                type="email"
              />
            </div>
            <div
              class="login__input-wrapper form__input-wrapper"
            >
              <label
                class="visually-hidden"
              >
                Password
              </label>
              <input
                class="login__input form__input"
                name="password"
                placeholder="Password"
                required=""
                type="password"
              />
            </div>
            <button
              class="login__submit form__submit button"
              type="submit"
            >
              Sign in
            </button>
          </form>
        </section>
        <section
          class="locations locations--login locations--current"
        >
          <div
            class="locations__item"
          >
            <a
              class="locations__item-link"
              href="/"
            >
              <span>
                Amsterdam
              </span>
            </a>
          </div>
        </section>
      </div>
    </main>
  </div>
</div>
`;

exports[`Test routing Render main page when user navigate to "/" url 1`] = `
<div>
  <p
    style="color: rgb(68, 129, 195); text-align: center; font-weight: bolder; font-size: 50px; padding-top: 150px;"
  >
    Loading...
  </p>
</div>
`;
